meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh

```diff
@@ -108,12 +106,15 @@ function log {
 }
 
 function runPreHacks {
+    local _boot_mountpoint="$(grep $(blkid | grep resin-boot | cut -d ":" -f 1) /proc/mounts | cut -d ' ' -f 2)"
+
     # we might need to repartition this so make sure it is unmounted
     log "Make sure resin-boot is unmounted..."
-    if [ -z $BOOT_MOUNTPOINT ]; then
-        log WARN "Mount point for resin-boot partition could not be found. Probably is already unmounted."
-    else
-        umount $BOOT_MOUNTPOINT &> /dev/null
+    if [ -z $_boot_mountpoint ]; then
+        log WARN "Mount point for resin-boot partition could not be found. It is probably already unmounted."
+    # XXX support old devices
```

Comment:
A little more detailed comment here. As old devices had the boot mounted in /boot and the conf in conf partition.